### ISSUE TITLE
[xenial] Changing channels doesn't work as intended & "Try again" button fails

### ISSUE THREAD

**Issue by advocatux** [ubports/ubuntu-touch#693]
- Device: BQ E4.5 (krillin)
- Channel: RC
- Build: xenial RC1

### Steps to reproduce
I installed RC1 successfully yesterday, and then I tried to change to devel channel

### Expected behavior
Changing channels should work flawlessly

### Actual behavior
Changing release channels on _System Settings > Updates > Update Settings > Channels_ doesn't go smoothly because the info is not refreshed on the page. For example, the phone is on RC channel and if I select Devel channel, that change doesn't show when going back to the previous page. Well, that's a bit annoying but that's all. **The issue is**, once Devel is selected, the download failed and the "Try again" button (Volver a intentar) didn't work.

After rebooting the phone, I succeeded changing from RC to devel that time.

### Logfiles and additional information
This is a screenshot of the download step failing.

![channelchange](https://user-images.githubusercontent.com/30233652/41646460-28416090-7474-11e8-8c54-9bc71697d0fb.jpg)

**Comment by Arc676** [comment]
@advocatux Yes, can confirm the issue I opened is similar to the second part of this issue.

On my device it says "canceled" instead because I didn't switch channels, but the behavior is the same.

**Comment by NeoTheThird** [comment]
It's not a xenial issue @mariogrip. Do you still want to take this in ota-4?

**Comment by kwah-gh** [comment]
Still observe this on OTA-5. And it is very confusing. From user perspective UI hangs on the way back from page with channel switcher (update settings?) to updates page. It looks like upon channel switch a lot is happening behind the scenes like actual check for availability of updates and download is started, when system update is available. At least, by the time I was able to get back to updates screen after stable -> rc switch, download of rc-image was almost complete.
